src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/CoscPaidInFullApplicantNotificationHandler.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.civil.handler.callback.camunda.notification;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+import uk.gov.hmcts.reform.civil.callback.Callback;
+import uk.gov.hmcts.reform.civil.callback.CallbackHandler;
+import uk.gov.hmcts.reform.civil.callback.CallbackParams;
+import uk.gov.hmcts.reform.civil.callback.CaseEvent;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.notify.NotificationService;
+import uk.gov.hmcts.reform.civil.notify.NotificationsProperties;
+import uk.gov.hmcts.reform.civil.service.FeatureToggleService;
+import java.util.List;
+import java.util.Map;
+import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_APPLICANT_SOLICITOR1_FOR_PAID_IN_FULL_COSC;
+import static uk.gov.hmcts.reform.civil.utils.PartyUtils.buildClaimantReference;
+
+@Service
+@RequiredArgsConstructor
+public class CoscPaidInFullApplicantNotificationHandler extends CallbackHandler
+    implements NotificationData {
+
+    private static final List<CaseEvent> EVENTS = List.of(NOTIFY_APPLICANT_SOLICITOR1_FOR_PAID_IN_FULL_COSC);
+    public static final String TASK_ID = "NotifyApplicantSolicitorCoscApplication";
+    private static final String REFERENCE_TEMPLATE = "cosc-application-email-notification-%s";
+
+    private final NotificationService notificationService;
+    private final NotificationsProperties notificationsProperties;
+    private final FeatureToggleService featureToggleService;
```

Comment:
Do we need this field featureToggleService?

Reply:
updated